beacon-chain/sync/rpc_send_request.go

```diff
@@ -198,6 +205,50 @@ func SendBlobSidecarByRoot(
 
 type blobResponseValidation func(blocks.ROBlob) error
 
+func chainBlobValidations(vf ...blobResponseValidation) blobResponseValidation {
+	return func(blob blocks.ROBlob) error {
+		for i := range vf {
+			if err := vf[i](blob); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+}
+
+type seqBlobValid struct {
+	prev *blocks.ROBlob
+}
+
+func (sbv *seqBlobValid) nextValid(blob blocks.ROBlob) error {
+	if sbv.prev == nil {
+		sbv.prev = &blob
+		return nil
+	}
+	if sbv.prev.BlockRoot() == blob.BlockRoot() {
+		// blob indices in responses should be strictly monotonically incrementing
+		if blob.Index != sbv.prev.Index+1 {
+			return errChunkResponseIndexNotAsc
+		}
+		if blob.Index >= fieldparams.MaxBlobsPerBlock {
+			return ErrBlobIndexOutOfBounds
+		}
+	} else {
+		if sbv.prev.Slot() >= blob.Slot() {
+			return errChunkResponseSlotNotAsc
```

Comment:
Before this, we should check that the `prev.BlockRoot()` matches `blob.ParentBlockRoot()`. It's a cheap check that ensures it's a connected chain 😄

Reply:
done